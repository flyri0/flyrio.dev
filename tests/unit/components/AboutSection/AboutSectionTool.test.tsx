import { render } from '@testing-library/react';
import { MdMoney } from 'react-icons/md';
import AboutsectionTool, {
  AboutSectionToolProps,
} from '@components/AboutSection/AboutSectionTool';

describe('<AboutSectionTool />', () => {
  it('should render the component', () => {
    const component = render(
      <AboutsectionTool inlineText="Money">
        <MdMoney />
      </AboutsectionTool>
    );

    expect(component).toBeTruthy();
  });

  it('should match snapshot', () => {
    const component = render(
      <AboutsectionTool inlineText="Money">
        <MdMoney />
      </AboutsectionTool>
    );

    expect(component).toMatchSnapshot();
  });
});
