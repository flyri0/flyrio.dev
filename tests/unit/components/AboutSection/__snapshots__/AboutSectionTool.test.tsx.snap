// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<AboutSectionTool /> > should match snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="flex cursor-default flex-row items-center gap-x-2 font-spaceGrotesk font-bold transition hover:-translate-y-0.5 hover:text-highlight"
      >
        <svg
          fill="currentColor"
          height="30px"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 24 24"
          width="30px"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M0 0h24v24H0V0z"
            fill="none"
          />
          <path
            d="M5 8h2v8H5zm7 0H9c-.55 0-1 .45-1 1v6c0 .55.45 1 1 1h3c.55 0 1-.45 1-1V9c0-.55-.45-1-1-1zm-1 6h-1v-4h1v4zm7-6h-3c-.55 0-1 .45-1 1v6c0 .55.45 1 1 1h3c.55 0 1-.45 1-1V9c0-.55-.45-1-1-1zm-1 6h-1v-4h1v4z"
          />
          <path
            d="M4 6h16v12H4z"
            fill="none"
          />
          <path
            d="M2 4v16h20V4H2zm2 14V6h16v12H4z"
          />
        </svg>
         
        <p>
          Money
        </p>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="flex cursor-default flex-row items-center gap-x-2 font-spaceGrotesk font-bold transition hover:-translate-y-0.5 hover:text-highlight"
    >
      <svg
        fill="currentColor"
        height="30px"
        stroke="currentColor"
        stroke-width="0"
        viewBox="0 0 24 24"
        width="30px"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M0 0h24v24H0V0z"
          fill="none"
        />
        <path
          d="M5 8h2v8H5zm7 0H9c-.55 0-1 .45-1 1v6c0 .55.45 1 1 1h3c.55 0 1-.45 1-1V9c0-.55-.45-1-1-1zm-1 6h-1v-4h1v4zm7-6h-3c-.55 0-1 .45-1 1v6c0 .55.45 1 1 1h3c.55 0 1-.45 1-1V9c0-.55-.45-1-1-1zm-1 6h-1v-4h1v4z"
        />
        <path
          d="M4 6h16v12H4z"
          fill="none"
        />
        <path
          d="M2 4v16h20V4H2zm2 14V6h16v12H4z"
        />
      </svg>
       
      <p>
        Money
      </p>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
